webapp/src/js/services/resource-icons.js

```diff
@@ -106,8 +106,12 @@ angular.module('inboxServices').factory('ResourceIcons',
         return DB().get(doc).then(res => Object.keys(res.resources));
       },
       getAppTitle: () => DB().get(DOC_IDS[1]).then(doc => doc.title),
-      replacePlaceholders: $elem => {
-        updateResources(DOC_IDS[0]).then(() => updateDom($elem, DOC_IDS[0]));
+      replacePlaceholders: ($elem, bypassCache = false) => {
+        if (!bypassCache && cache[DOC_IDS[0]].doc) {
+          updateDom($elem, DOC_IDS[0]);
+        } else {
+          updateResources(DOC_IDS[0]).then(() => updateDom($elem, DOC_IDS[0]));
+        }
```

Comment:
I don't think this parameter is particularly useful, no code uses it and the cache is automatically updated every time a relevant change is received. 
In the past, the way this function worked was: a reference to a the promise that initially loaded the resources was saved and used in `replacePlaceholders` call. The `get` request would only be called once, as the promise would be resolved for every other subsequent call. 
With the current implementation, even with this fix of checking cache, you could end up requesting the `resources` doc multiple times if multiple `replacePlaceholders` calls are made in quick succession, while cache is not yet set.